nova_vm/src/ecmascript/abstract_operations/operations_on_iterator_objects.rs

```diff
@@ -0,0 +1,302 @@
+//! ## [7.4 Operations on Iterator Objects](https://tc39.es/ecma262/#sec-operations-on-iterator-objects)
+
+use super::{
+    operations_on_objects::{call, get},
+    type_conversion::{to_boolean, to_object},
+};
+use crate::{
+    ecmascript::{
+        abstract_operations::operations_on_objects::get_method,
+        execution::{
+            agent::{ExceptionType, JsError},
+            Agent, JsResult,
+        },
+        types::{Function, Object, OrdinaryObject, PropertyKey, Value},
+    },
+    heap::WellKnownSymbolIndexes,
+};
+
+/// [7.4.2 GetIteratorFromMethod ( obj, method )](https://tc39.es/ecma262/#sec-getiteratorfrommethod)
+///
+/// The abstract operation GetIteratorFromMethod takes arguments obj (an
+/// ECMAScript language value) and method (a function object) and returns
+/// either a normal completion containing an Iterator Record or a throw
+/// completion.
+pub(crate) fn get_iterator_from_method(
+    agent: &mut Agent,
+    obj: Value,
+    method: Function,
```

Comment:
**issue:** When the spec says "iterator record" it does not refer to an object of a given form, not as far as I can tell anyhow.

It refers to a particular struct form, essentially, which contains the fields described [here](https://tc39.es/ecma262/#sec-iterator-records). So in effect this should make it much easier to implement some of these steps, eg. steps 4 and 5 here are just returning a struct.

Reply:
Oh, ok. Was planning on asking in the discord when I had time but seems like that wont be needed, thanks! Where should I define this struct? I guess it belongs in this file considering it was defined in this section of the spec?

Reply:
Yeah. It seems like a stupid place to define it but since this is the place in the spec then let's follow it.

It might turn out to be a good place to define it.